src/planner/bind_context.cpp

```diff
@@ -335,6 +335,9 @@ void BindContext::GenerateAllColumnExpressions(StarExpression &expr,
 				if (CheckExclusionList(expr, column_name, new_select_list, excluded_columns)) {
 					continue;
 				}
+				if (CheckExclusionList(expr, binding.alias + "." + column_name, new_select_list, excluded_columns)) {
```

Comment:
I think it might be easier to create a ColumnRefExpression here, then change `CheckExclusionList`
We'll loop through the `exclude_list` of the StarExpression (which we'll also turn into a unordered_set of ColumnRefExpressions)

If the expression from the `exclude_list` is qualified (`IsQualified()`) then we'll check the qualifier of the column ref expression we created here, otherwise it's ignored.

Reply:
I think it makes sense to create a `columnref_set_t` (have a look at `case_insensitive_map.hpp` to see what this would look like), it's essentially a `unordered_set<ColumnRefExpression>` with custom equality+hash methods